models/issue.go

```diff
@@ -1568,25 +1581,24 @@ func SearchIssueIDsByKeyword(kw string, repoIDs []int64, limit, start int) (int6
 	return total, ids, nil
 }
 
-func updateIssue(e Engine, issue *Issue) error {
-	_, err := e.ID(issue.ID).AllCols().Update(issue)
-	if err != nil {
-		return err
-	}
-	return nil
+func updateIssueByCols(e Engine, issue *Issue, columns ...string) (err error) {
+	_, err = e.ID(issue.ID).Cols(columns...).Update(issue)
+	return
 }
 
-// UpdateIssue updates all fields of given issue.
-func UpdateIssue(issue *Issue) error {
+// UpdateIssueByAPI updates all allowed fields of given issue.
+func UpdateIssueByAPI(issue *Issue) error {
+	// allowed fields to update
+	columns := []string{"name", "is_closed", "content", "milestone_id", "priority",
+		"deadline_unix", "updated_unix", "closed_unix", "is_locked"}
```

Comment:
I see what you're trying to do - but I think we could be making problems for the future here. Is there really no easy of telling xorm not to update one particular column? I thought if it was set to the default it might not do it?

Reply:
@zeripath see Lunny's comments: [link1](https://github.com/go-gitea/gitea/pull/9539#pullrequestreview-336986762) and [link2](https://github.com/go-gitea/gitea/pull/9539#pullrequestreview-336986876)

there is more than one colum to protect